File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        List<int> num_stack = new List<int>();
        int left_val = 0;
        int right_val = 0;

        // for i in tokens:
        //     if i == "+":
        //         left_val = num_stack.pop()
        //         right_val = num_stack.pop()
        //         num_stack += [right_val + left_val]
        //     elif i == "-":
        //         left_val = num_stack.pop()
        //         right_val = num_stack.pop()
        //         num_stack += [right_val - left_val]
        //     elif i == "*":
        //         left_val = num_stack.pop()
        //         right_val = num_stack.pop()
        //         num_stack += [right_val * left_val]
        //     elif i == "/":
        //         left_val = num_stack.pop()
        //         right_val = num_stack.pop()
        //         num_stack += [right_val / left_val]
        //     else:
        //         num_stack.append(int(i))
        
        // return int(num_stack[-1])

        foreach(string token in tokens){
            if (token == "+"){
                left_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                right_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                num_stack.Add(right_val + left_val);
            } 
            else if (token == "-"){
                left_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                right_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                num_stack.Add(right_val - left_val);
            }
            else if (token == "*"){
                left_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                right_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                num_stack.Add(right_val*left_val); 
            }
            else if (token == "/"){
                left_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                right_val = num_stack[num_stack.Count - 1];
                num_stack.RemoveAt(num_stack.Count - 1);

                num_stack.Add(right_val/left_val);
            }
            else {
                num_stack.Add(int.Parse(token));
            }
        }

        return(num_stack[num_stack.Count - 1]);
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        s = list(s)
        str_openings_stack = []

        for char in s:
            if (char == '(') or (char == '{') or (char == '['):
               str_openings_stack.append(char)
            else:
                if str_openings_stack:
                    if char == ')':
                        if str_openings_stack[-1] == '(':
                            str_openings_stack.pop()
                            continue
                        else:
                            return False
                    elif char == '}':
                        if str_openings_stack[-1] == '{':
                            str_openings_stack.pop()
                            continue
                        else:
                            return False
                    elif char == ']':
                        if str_openings_stack[-1] == '[':
                            str_openings_stack.pop()
                            continue
                        else:
                            return False
                else:
                    return False
        
        if str_openings_stack:
            return False
        else:
            return True